ansible_collections/arista/avd/plugins/plugin_utils/eos_designs_shared_utils/wan.py

```diff
@@ -191,13 +191,14 @@ def get_public_ip_for_wan_interface(self: SharedUtils, interface: dict) -> str:
         """
         Takes a dict which looks like `l3_interface` from node config
 
-        If not a WAN route-server this just returns the interface IP.
+        If not a WAN route-server this returns public IP and if not found then the interface IP.
 
         For WAN route-servers we try to find the IP under wan_route_servers.path_groups.interfaces.
-        If not found we use the IP under the interface, unless it is "dhcp" where we raise.
+        If not found we look for the public_ip and then the ip_address under the interface.
+        If there is no public_ip and if ip_address is "dhcp" we raise an error.
         """
         if not self.is_wan_server:
-            return interface["ip_address"]
+            return default(interface.get("public_ip"), interface["ip_address"])
```

Comment:
@gmuloc It should not really be needed to get the public_ip for a client right? I don't think it will hurt, so we can keep this.

Reply:
as of today this is not used indeed so not hurting and not needed. however it is more correct that this function returns the public_ip on the WAN interface if set rather than the ip_address so I would leave it like this for now